### DISCUSSION TITLE
Sonoff dongle P. coordinator. Do I need to update the firmware

### DISCUSSION THREAD

**Discussion by upandabout** [Koenkk/zigbee2mqtt#21861] [Devices]
Hi All.

So I'm brand new, to HA and Zigbee2mqtt. I managed to get HA and zigbee2mqtt running and working on a Pi4B 4GB with a Sonoff dongle P coordinator.
The Sonoff dongle P was purchased just before Christmas and seams to be working. I did try to update the firmware, but the boot button on the dongle P would not let me enter the boot loader, so I just tried to se if it would work out of the box, and it does. Now I could just continue to setup my whole zigbee network but then tought that If I'm going to update it later because I didn't bother now I would probably end up hating myself for not doing it before really getting startet with setting up all the devices and the zigbee network, as it might become a hassel If I need to find, reset and repair all the devices afterwards du to coordinator update.

Q1. So I wanted to check what firmware is actually on the device, but I have no clue on how to find that info in the HA or zigbee2mqtt addon software?
Q2. If I need to update it, anybody got a hint on how to update it, the smart flash programmer software is not working due to Sonoff blocking the load button on the device or the button is out of service...?
Q3. I think remember seeing something about turning up the transmitting power of the dongle, how is that achieved and is it needed? 

Thanks in advance, your help is much appreciated. /Kaseper

**Comment by upandabout** [comment]
So no help, I'm still on the original firmware, do I now need to update it?

**Reply by MSL-DA** [reply]
You are best to answer that yourself. Aren't you?

**Reply by MSL-DA** [reply]
> So no help, I'm still on the original firmware, do I now need to update it?

What do you want with your question? We can easily tell you how to update your "dongle", but there are countless guides on this topic. And again, what do you want to achieve? If you have a stable zigbee network then be happy with that.
